proctor-common/src/main/java/com/indeed/proctor/common/Proctor.java

```diff
@@ -515,4 +523,56 @@ private boolean isIncognitoEnabled(@Nonnull final Map<String, Object> inputConte
                 .map(value -> value.equalsIgnoreCase("true"))
                 .orElse(false);
     }
+
+    /**
+     * Aggregates properties of Payload Experiments for look up in Proctor Result. Checks Priority
+     * of Property to handle conflicting overrides of properties within namespaces. Properties are
+     * the Key:Value pairs of fields stored in JsonNode Payload type.
+     */
+    private void populateProperties(
+            final String testName,
+            final TestBucket testBucket,
+            final Map<String, PayloadProperty> testProperties) {
+        final Payload payload = testBucket.getPayload();
+        if (payload != null && payload.getJson() != null) {
+            payload.getJson()
+                    .fields()
+                    .forEachRemaining(
+                            field -> attemptStoringProperty(field, testName, testProperties));
+        }
+    }
+
+    private void attemptStoringProperty(
+            final Map.Entry<String, com.fasterxml.jackson.databind.JsonNode> field,
+            final String testName,
+            final Map<String, PayloadProperty> testProperties) {
+        final PayloadProperty curr = testProperties.get(field.getKey());
+        // store property if it does not exist in map
+        if (curr == null) {
+            testProperties.put(
+                    field.getKey(),
+                    PayloadProperty.builder().value(field.getValue()).testName(testName).build());
+        } else {
+            final PayloadExperimentConfig currPayloadConfig =
+                    testChoosers
```

Comment:
nit: should we extract common code to 
getPayloadExperimentConfigFromTestDefinition(testName)?

Reply:
I personally do not think it is worth a method. Not that much duplication